registry/registry.go

```diff
@@ -89,9 +89,25 @@ func WrapTransport(transport http.RoundTripper, url, username, password string)
 	return errorTransport
 }
 
-func newFromTransport(registryUrl, username, password string, transport http.RoundTripper, logf LogfCallback) (*Registry, error) {
+func WrapTokenTransport(transport http.RoundTripper, url, username, password string) http.RoundTripper {
+	tokenTransport := &TokenTransport{
+		Transport: transport,
+		Username:  username,
+		Password:  password,
+	}
+	errorTransport := &ErrorTransport{
+		Transport: tokenTransport,
+	}
+	return errorTransport
+}
+
+func newFromTransport(registryUrl, username, password string, transport http.RoundTripper, disableBasicAuth bool, logf LogfCallback) (*Registry, error) {
 	url := strings.TrimSuffix(registryUrl, "/")
-	transport = WrapTransport(transport, url, username, password)
+	if disableBasicAuth == true {
```

Comment:
if you agree with the suggestion above, this (106-110) could be replaced with a call to `wrapTransport`.